From 031b4fc70be6f50f7a3e1d75a37872b9e44cf9a4 Mon Sep 17 00:00:00 2001
From: Rob Clark <robdclark@gmail.com>
Date: Sun, 6 Aug 2017 14:10:07 -0400
Subject: [PATCH 1/1] efi_loader: LocateHandle should return EFI_NOT_FOUND if
 none found

Spotted this debugging OpenBSD's bootloader in qemu.  (Wouldn't really
fix anything, the problem was not having any disks, but we should
probably return the correct error code.)

Signed-off-by: Rob Clark <robdclark@gmail.com>
---
 lib/efi_loader/efi_boottime.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 1a747a5b68..a924e45096 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -633,6 +633,10 @@ static efi_status_t EFIAPI efi_locate_handle(
 		return EFI_BUFFER_TOO_SMALL;
 	}
 
+	*buffer_size = size;
+	if (size == 0)
+		return EFI_NOT_FOUND;
+
 	/* Then fill the array */
 	list_for_each(lhandle, &efi_obj_list) {
 		struct efi_object *efiobj;
@@ -642,7 +646,6 @@ static efi_status_t EFIAPI efi_locate_handle(
 		}
 	}
 
-	*buffer_size = size;
 	return EFI_SUCCESS;
 }
 
-- 
2.11.0

